### ISSUE TITLE
'ERROR: Job failed: exit status 1' appears when I run sonar through maven execution 

### ISSUE THREAD

**Issue by he3naika** [gabrie-allaigre/sonar-gitlab-plugin#68]
**Gitlab version:** 10.0.1
**Gitlab installed on** Linux 3.10.0-327.el7.x86_64
**Gitlab sonar plugin version:** 2.1.0
**SonarQube installed on** Ubuntu 3.11.0-26-generic

See '.gitlab-ci.yml' below
```ruby
sonarqube_preview:
  script:
    - git config --global user.email "jenkins@test.com"
    - git config --global user.name "jenkins"
    - git checkout origin/development
    - git merge $CI_COMMIT_SHA --no-commit --no-ff
# For changes in Merge Request
    - mvn --batch-mode clean install sonar:sonar -Dmigration.skip=true -Dsonar.host.url=http://SONAR_URL:PORT/ -Dsonar.login=SONAR_LOGIN -Dsonar.password=SONAR_PASSWORD -Dsonar.analysis.mode=preview -Dsonar.gitlab.commit_sha=$CI_COMMIT_SHA -Dsonar.gitlab.ref_name=$CI_COMMIT_REF_NAME -Dsonar.gitlab.project_id=$CI_PROJECT_ID -Dsonar.sourceEncoding=UTF-8 -Dsonar.language=java -Dsonar.projectKey=SONAR_PROJECT_KEY -Dsonar.profile=SONAR_PROFILE -Dsonar.java.source=1.8 -Dsonar.projectName=SONAR_PROJECTNAME
  stage: test
  except:
    - master
    - development
    - /^hotfix_.*$/
    - /.*-hotfix$/
    - tags
  tags:
    - TEMP

sonarqube:
  script:
    - mvn --batch-mode clean install sonar:sonar -Dmigration.skip=true -Dsonar.host.url=http://SONAR_URL:PORT/ -Dsonar.login=SONAR_LOGIN -Dsonar.password=SONAR_PASSWORD -Dsonar.sourceEncoding=UTF-8 -Dsonar.language=java -Dsonar.projectKey=SONAR_PROJECT_KEY  -Dsonar.profile=SONAR_PROFILE -Dsonar.java.source=1.8 -Dsonar.projectName=SONAR_PROJECTNAME
  stage: test
  only:
    - development
  tags:
    - TEMP
```

See `Pipeline Job log` below
```log
[INFO] 447 files indexed
[INFO] Quality profile for java: TEMP_Profile
[INFO] Sensor JavaSquidSensor [java]
[INFO] Configured Java source version (sonar.java.source): 8
[INFO] JavaClasspath initialization
[INFO] JavaClasspath initialization (done) | time=38ms
[INFO] JavaTestClasspath initialization
[INFO] JavaTestClasspath initialization (done) | time=4ms
[INFO] Java Main Files AST scan
[INFO] 429 source files to be analyzed
[INFO] 30/429 files analyzed, current file: /home/gitlab-runner/builds/82c2eccc/0/TEMP/TEMP/TEMP-ui/app/controllers/Application.java
[INFO] 34/429 files analyzed, current file: /home/gitlab-runner/builds/82c2eccc/0/TEMP/TEMP/TEMP-ui/app/controllers/Employees.java
[INFO] 36/429 files analyzed, current file: /home/gitlab-runner/builds/82c2eccc/0/TEMP/TEMP/TEMP-ui/app/controllers/Holidays.java
[INFO] 74/429 files analyzed, current file: /home/gitlab-runner/builds/82c2eccc/0/TEMP/TEMP/TEMP-ui/app/controllers/people/PersonGeneralInfoEdit.java
[INFO] 101/429 files analyzed, current file: /home/gitlab-runner/builds/82c2eccc/0/TEMP/TEMP/TEMP-ui/app/controllers/projects/WorkOrders.java
[INFO] 256/429 files analyzed, current file: /home/gitlab-runner/builds/82c2eccc/0/TEMP/TEMP/TEMP-ui/app/models/holder/reports/DaysInMonthHolder.java
[INFO] 378/429 files analyzed, current file: /home/gitlab-runner/builds/82c2eccc/0/TEMP/TEMP/TEMP-ui/app/persistence/PersonProjectRepository.java
[INFO] 385/429 files analyzed, current file: /home/gitlab-runner/builds/82c2eccc/0/TEMP/TEMP/TEMP-ui/app/persistence/ProjectReviewRepository.java
[INFO] 385/429 files analyzed, current file: /home/gitlab-runner/builds/82c2eccc/0/TEMP/TEMP/TEMP-ui/app/persistence/ProjectReviewRepository.java
[INFO] 385/429 files analyzed, current file: /home/gitlab-runner/builds/82c2eccc/0/TEMP/TEMP/TEMP-ui/app/persistence/ProjectReviewRepository.java
[INFO] 385/429 files analyzed, current file: /home/gitlab-runner/builds/82c2eccc/0/TEMP/TEMP/TEMP-ui/app/persistence/ProjectReviewRepository.java
[INFO] 385/429 files analyzed, current file: /home/gitlab-runner/builds/82c2eccc/0/TEMP/TEMP/TEMP-ui/app/persistence/ProjectReviewRepository.java
bash: line 54: 14887 Killed                  - mvn --batch-mode clean install sonar:sonar -Dmigration.skip=true -Dsonar.host.url=http://SONAR_URL:PORT/ -Dsonar.login=SONAR_LOGIN -Dsonar.password=SONAR_PASSWORD -Dsonar.analysis.mode=preview -Dsonar.gitlab.commit_sha=$CI_COMMIT_SHA -Dsonar.gitlab.ref_name=$CI_COMMIT_REF_NAME -Dsonar.gitlab.project_id=$CI_PROJECT_ID -Dsonar.sourceEncoding=UTF-8 -Dsonar.language=java -Dsonar.projectKey=SONAR_PROJECT_KEY -Dsonar.profile=SONAR_PROFILE -Dsonar.java.source=1.8 -Dsonar.projectName=SONAR_PROJECTNAME
ERROR: Job failed: exit status 1
```
I didn't find any info about exception in all possible log files (GitLab, SonarQube etc)

**Comment by gabrie-allaigre** [comment]
Hi,
Only crash with sonar preview or full analyse ?
Add debug mode for full log.

**Comment by he3naika** [comment]
Hi,
Crash for 'preview' and 'issues' analysis modes. 'publish' analysis mode the GitLab plugin is not supported.
Full log (get it add -X option to mvn execution) for both modes you can download via links below:
[sonar.analysis.mode=preview](https://www.dropbox.com/s/ooat7wu6zbgug6g/FULL%20LOG_sonar_analysis_mode%3Dpreview.txt?dl=0)
[sonar.analysis.mode=issues](https://www.dropbox.com/s/z67uxv9d13bqe9t/FULL%20LOG_sonar_analysis_mode%3Dissues.txt?dl=0)

**Comment by gabrie-allaigre** [comment]
does the preview work without activating the gitlab plugin ? (no set commit_sha)

**Comment by he3naika** [comment]
Yes. ```sonar.analysis.mode=preview``` works without activating the gitlab plugin
You can [donwload](https://www.dropbox.com/s/69ffwgijlcted1t/preview%20works%20without%20activating%20the%20gitlab%20plugin.txt?dl=0) and see execution log

**Comment by gabrie-allaigre** [comment]
Sorry, but last test not running with Maven ?

**Comment by he3naika** [comment]
Last test was running with Maven too. I've publisted log without maven part. Only qube part

**Comment by gabrie-allaigre** [comment]
Last log use sonar-scanner not maven

```
/var/lib/jenkins/tools/hudson.plugins.sonar.SonarRunnerInstallation/SonarQubeRunner/bin/sonar-scanner -e -Dsonar.host.url=http://SONAR_URL/ ******** -Dsonar.sourceEncoding=UTF-8 -Dsonar.sources=bl-ear,TEMP-model,TEMP-parent,TEMP-security-model,TEMP-ui -Dsonar.language=java -Dsonar.exclusions=TEMP-ui/modules/**/* -Dsonar.projectVersion=297 -Dsonar.java.binaries=**/target/classes -Dsonar.projectKey=Java:TEMP -Dsonar.profile=TEMP_Profile -Dsonar.java.source=1.8 -Dsonar.analysis.mode=preview -Dsonar.projectName=TEMP -Dsonar.projectBaseDir=/var/lib/jenkins/jobs/MasterStaticCodeAnalysis/workspace
```

**Comment by he3naika** [comment]
Sorry. It is my fault. I have published incorrect log file.
Please [download](https://www.dropbox.com/s/soefo6wkh9kyt09/preview%20works%20without%20activating%20the%20gitlab%20plugin.txt?dl=0) and see new one.

**Comment by gabrie-allaigre** [comment]
Sorry, I have not too much idea about the problem.
Does it also do the same thing with sonar-scanner?

**Comment by he3naika** [comment]
Sorry, i dont understand your last question for sonar-scanner. Could you extend it?

**Comment by gabrie-allaigre** [comment]
sorry, if you run with sonar-scanner instead of maven, there is the same problem with the plugin gitlab?

**Comment by he3naika** [comment]
sonar-scaner + gitlab plugin works well. Execution log is available via [link](https://www.dropbox.com/s/azcf2dhe2pyufmq/LOG_sonar-scaner_sonar_analysis_mode%3Dpreview.txt?dl=0)
As I see after investigation, does not work maven + gitlab plugin configuration. Another configurations are work well.

**Comment by gabrie-allaigre** [comment]
Add `MAVEN_OPTS` env variable, and more memory needs to be allocated `-Xmx2048M`

**Comment by he3naika** [comment]
Why did you desite that is "java.lang.OutOfMemoryError: Java heap space" problem?
I didn't find "OutOfMemoryError" in any log files.

I have added "-Xmx2048M" to maven execution ```mvn --batch-mode clean install sonar:sonar -DargLine=-Xmx2048m``` but ```ERROR: Job failed: exit status 1``` still appiars.

**Comment by gabrie-allaigre** [comment]
You always have to end bash: line 54: 14887 Killed ?

**Comment by he3naika** [comment]
No. Any runs I have a different lines. A common for every run this is a fail on step where source files are analyzing ```426 source files to be analyzed```

**Comment by gabrie-allaigre** [comment]
In previous log, at the end there is the maven process that is killed by linux.

https://stackoverflow.com/questions/7278514/maven-compilation-dies-with-killed

you can show me the last log with -Xmx2048m ?

**Comment by he3naika** [comment]
This log you can find via [link](https://www.dropbox.com/s/zt1uedvu3vqybd5/mvn_Xmx2048m.txt?dl=0)

**Comment by he3naika** [comment]
Issue was fixed after update SonarQube to 6.7.0.33306 version.
Current issue can be closed.
